src/components/abstractComponent.ts

```diff
@@ -0,0 +1,436 @@
+/**
+ * Copyright 2014-present Palantir Technologies
+ * @license MIT
+ */
+
+import * as d3 from "d3";
+
+import { ComponentContainer } from "./componentContainer";
+
+import { Point, SpaceRequest, Bounds } from "../core/interfaces";
+import * as RenderController from "../core/renderController";
+import * as Utils from "../utils";
+import { Set } from "../utils/set";
+
+export type GenericComponentCallback<D> = (component: IComponent<D>) => void;
```

Comment:
Also the callback function below is a function but uses the interface "I.." prefix. Make these consistent with eachother and other function definitions. I'm not sure if prefix or no prefix is better.